File: src/main/org/nlogo/prim/etc/_worldwidth.java
package org.nlogo.prim.etc;

import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.api.Syntax;

public final strictfp class _worldwidth extends Reporter {
  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax(Syntax.NumberType());
  }

  @Override
  public Object report(Context context) {
    return report_1(context);
  }

  public Double report_1(Context context) {
    return world.worldWidthBoxed();
  }

  public double report_2(Context context) {
    return world.worldWidth();
  }
}

File: src/main/org/nlogo/prim/etc/_worldwidth.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Reporter }

class _worldwidth extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(Syntax.NumberType)
  override def report(context: Context) =
    report_1(context)
  def report_1(context: Context): java.lang.Double =
    world.worldWidthBoxed
  def report_2(context: Context): Double =
    world.worldWidth
}
